### PR TITLE
Create Github Action that deploys the hello-strongbox-npm into the Github Package Registry

### PR DESCRIPTION
**Pull request by j4ckofalltrades**
Refs strongbox/strongbox#1927

### CODE REVIEW

**Comment by j4ckofalltrades** [comment]
Key thing here is that _name_ field in the _package.json_ name field should should match the GitHub repository name.
From [docs](https://docs.github.com/en/free-pro-team@latest/actions/guides/publishing-nodejs-packages):

> By default, GitHub Packages publishes a package in the GitHub repository you specify in the name field of the package.json file. For example, you would publish a package named @my-org/test to the my-org/test GitHub repository. For more information, see npm-scope in the npm documentation.

Example commit: https://github.com/j4ckofalltrades/strongbox-examples/commit/4279e0adc5404fdbc34b740fac21944f83664398
Example package: https://github.com/j4ckofalltrades/strongbox-examples/packages/468529

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+          node-version: '12.x'
+          registry-url: 'https://npm.pkg.github.com'
+          # Defaults to the user or organization that owns the workflow file
+          scope: '@strongbox'
+      - run: npm install
+        working-directory: 'hello-strongbox-npm'
+      - run: npm publish
+        working-directory: 'hello-strongbox-npm'
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (j4ckofalltrades) [inline .github/workflows/main.yml]: Jobs for deploying examples of other layout providers can just be appended here e.g.

```
publish-hello-strongbox-maven:
...
publish-hello-strongbox-gradle:
...
```

**Comment by j4ckofalltrades** [comment]
@carlspring Requesting review.

**Review by carlspring — APPROVED** [review]
LGTM! Thanks!

@steve-todorov : What are your thoughts on this?

**Review by steve-todorov — APPROVED** [review]
LGTM! The only small note I have is the file name should be called `gh-registry.yml` since this workflow will be only for this purpose.

**Comment by j4ckofalltrades** [comment]
Updated file name to `gh-registry.yml`.

**Comment by steve-todorov** [comment]
Works like a charm! Great job! :) 

Just one question to clarify things for myself. Do I need to set `GH_TOKEN` secret or GH Actions picks it up on it's own? 
I did add one, but wanted to check if you also added it.

**Comment by carlspring** [comment]
Excellent job, thank you! :)

**Comment by j4ckofalltrades** [comment]
@steve-todorov

> Do I need to set GH_TOKEN secret or GH Actions picks it up on it's own? I did add one, but wanted to check if you also added it.

GitHub provides it automatically for you.

https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow

**Comment by carlspring** [comment]
Cool! Thanks for the clarification! :)

**Pull request merged**